src/main/java/com/mraof/minestuck/alchemy/AlchemyHelper.java

```diff
@@ -85,6 +85,11 @@ else if(tag.contains("contentSize") && tag.getInt("contentSize") >= 1)
 		
 	}
 	
+	public static boolean isReadableCard(ItemStack readItem)
+	{
+		return readItem.is(MSItems.CAPTCHA_CARD.get()) && readItem.hasTag() && readItem.getTag().getBoolean("decoded");
```

Comment:
I recommend also allowing it to be readable if its already been punched. Both because that means the player can already reverse engineer the code via interpreting the punch pattern and because the part of the card with the captcha on it likely has holes punched through it

Reply:
done

Reply:
decided against it, although it make since functionally i will not be making this one function do two things like that and will instead just include the isPunched() function in all calls involving isReadable()

Reply:
decided against this as i'm tired and don't want to think about how to make that work

Reply:
return isPunchedCard(readItem) || (readItem.is(MSItems.CAPTCHA_CARD.get()) && readItem.hasTag() && readItem.getTag().getBoolean("decoded"));

Reply:
The above just builds the check into this boolean. If the item in question meets either the punched card criteria or the criteria you created, it will succeed

Reply:
i was referring to my first comment with my second comment

Reply:
Does this mean you think punched cards should still require decoding?